cluster-autoscaler/metrics/metrics.go

```diff
@@ -138,6 +138,22 @@ var (
 		},
 	)
 
+	nodesGroupMinNodes = k8smetrics.NewGaugeVec(
+		&k8smetrics.GaugeOpts{
+			Namespace: caNamespace,
+			Name:      "node_group_min_count",
+			Help:      "Minimum number of nodes in the node group",
+		}, []string{"node_group"},
```

Comment:
one issue here is that the cardinality of these metrics will essentially be unbounded. i have been advised against doing this type of thing in the past, see this discussion https://github.com/kubernetes/autoscaler/pull/3922#discussion_r586939682

Reply:
for the majority of users this should be totally ok. if a cluster is huge enough to have dozens of dozens of nodegroups, its Prometheus should either way handle another bunch of time series.

CA is a very usable component to expose such metrics and provide data for alerting for NodeGroups stuff

Reply:
i generally agree with you. i just wanted to highlight that i proposed something similar with respect to adding node group IDs, although in that case the metrics were re-calculated frequently. in this case they are just published once, so maybe that is more applicable.

i am curious those, what about node groups that get added after the autoscaler has started?

Reply:
>i am curious those, what about node groups that get added after the autoscaler has started?

i think it will discover them automatically if configured to do so via tags etc

Reply:
yeah, seems like new node groups would get picked up, but changes to the min/max would not.

Reply:
I understand your concern about cardinality. But node groups are bounded. Compare that to something like `kube_node_info` which has the node label.

Reply:
@mwielgus you had strong opinions on label cardinality in the past, could you weigh in here?

Reply:
I definitely argued against using identifiers as metric labels due to cardinality issues. I broadly disagree with the assumption that if something is useful but doesn't scale well it's fine since most users don't run clusters with thousands of nodes and so most users will benefit from it. And I disagree that number of nodegroups can be assumed to be small. 

That being said those metrics are obviously very specific to monitoring individual nodegroups. It's not really stopping anyone from using existing metrics in a large cluster - they can just not use the new metrics. So I'm generally fine with adding those new metrics. My issue is that prometheus will start collecting those metrics by default and you'll need to add relabel_configs to stop doing it. Someone upgrading CA to a newer version is likely to miss it. And that will also be true if we add more nodegroup labeled metrics in future.
For that reason I'd like to consider hiding it behind a CA flag (ex. --emit-per-nodegroup-metrics). That way a user could easily opt-in/out of all such metrics.

Reply:
@MaciekPytel I'm fine adding this behind a flag. But I'd like to understand more the cardinality scope here. Even for clusters with thousands of nodes, how many node groups are we talking about here? A hundred, a couple of hundred? Given that such metrics are static by nature, I'd like to understand how this can lead to cardinality explosion. Even if node groups come and go, in a single TSDB block you should not have a high cardinality. 

Again, kubernetes itself emits metrics with node name labels, which have far higher cardinality.

Reply:
i like the notion of adding a flag which would enable node group labels for some metrics. i think there are other metrics which could benefit from having the node group ID as a label and it would be nice to empower users who understand the operational load of adding this extra data to their metrics solutions.

Reply:
Re: cardinality issue - thousands of nodegroups can happen (ex. #4058 is an optimization specifically for such case). It's hard to imagine the number any higher given that kubernetes itself only supports 5k nodes I think. However, it's not uncommon for large organization to run dozens/hundreds of clusters and I think joining the metrics from multiple clusters is where the cardinality really becomes an issue. I do agree this specific metric is much less of an issue than other metrics, since it's going to be very static. I think having a flag is still the way to go if we want to add more metrics labeled by nodegroup. Once we have such flag I see little reason why we couldn't add more metrics like this. I'm less sure about adding labels to existing metrics - variable number of metrics on a label seems like an easy source of confusion. Instead I'd propose creating an equivalent metric with slightly different name and emitting one or the other based on the flag. WDYT?

Reply:
>  I'm less sure about adding labels to existing metrics - variable number of metrics on a label seems like an easy source of confusion. Instead I'd propose creating an equivalent metric with slightly different name and emitting one or the other based on the flag. WDYT?

you make a good point about the possible confusion for users. i'm not sure how i feel about creating multiple metrics (with and without the labels), it seems a little onerous but that might just be my initial reaction. in the absence of another option, it seems like it might the best path forward. i'm happy to hear more conversation on this topic though. thanks @MaciekPytel !

Reply:
Thanks @MaciekPytel for the elaboration. Sounds good. I'll add the flag and update the PR.